packages/web-react/src/components/Grid/__tests__/Grid.test.tsx

```diff
@@ -14,39 +14,114 @@ describe('Grid', () => {
   restPropsTest(Grid, 'div');
 
   it('should render text children', () => {
-    const dom = render(<Grid>Hello World</Grid>);
+    const text = 'Hello world';
 
-    const element = dom.container.querySelector('div') as HTMLElement;
-    expect(element.textContent).toBe('Hello World');
+    render(<Grid>{text}</Grid>);
+
+    const element = screen.getByText(text) as HTMLElement;
+
+    expect(element).toBeInTheDocument();
   });
 
   it('should have desktop cols classname', () => {
-    const dom = render(<Grid cols={{ desktop: 3 }} />);
+    const testId = 'grid-test-id';
```

Comment:
That's an interesting approach. This could be useful if you want to use the same component in different tests with different data-testid.  However, if you have a component that is always tested with the same data-testid, it may be useful to add data-testid directly to the component definition. But this approach is not incorrect.

Reply:
Wasn't sure how the function scope behaves like. But we talked that we wouldn't worried about that so I will move the variables above the `it` functions.